### PR TITLE
(Packable) Pack and unpack byte buffers faster

### PR DESCRIPTION
**Pull request by pvdrz**
This PR changes the `Packable` implementations of `Vec<T>`, `Box<[T]>`, `VecPrefix<T, B>` and `BoxedSlicePrefix<T, B>` in order to specialize them when `T` is equal to `u8`.

Before this changes. packable would do something like

```rust
for item in self.iter() {
     item.pack(packer)?;
}
```
to pack any byte buffer, meaning that each byte will trigger one write in the packer. With the changes proposed here. Byte buffers are packed and unpacked in a single read/write. Some benchmarks show a 6x performance improvement.

I ran the test suite of packable using miri given that this code relies on unsafe heavily, No issues were found.

### CODE REVIEW

File: packable/packable/src/packable/array.rs [hunk 1]
```diff
@@ -3,14 +3,21 @@
 
 use crate::{error::UnpackError, packer::Packer, unpacker::Unpacker, Packable};
 
+use core::any::TypeId;
 use core::mem::MaybeUninit;
```
Comment (thibault-martinez) [inline packable/packable/src/packable/array.rs:7]: Merge together

File: packable/packable/src/packable/box.rs [hunk 2]
```diff
@@ -6,6 +6,10 @@ extern crate alloc;
 use crate::{error::UnpackError, packer::Packer, unpacker::Unpacker, Packable};
 
 use alloc::boxed::Box;
+#[cfg(feature = "usize")]
+use alloc::{vec, vec::Vec};
+#[cfg(feature = "usize")]
+use core::any::TypeId;
```
Comment (thibault-martinez) [inline packable/packable/src/packable/box.rs:12]: Merge both usize

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: packable/packable/src/packable/box.rs (hunk 2 above)
Comment (pvdrz) [reply packable/packable/src/packable/box.rs:12]: wdym?

**Comment by thibault-martinez** [comment]
Why is a lot of format being changed when the CI wasn't complaining about it ?

**Comment by pvdrz** [comment]
> Why is a lot of format being changed when the CI wasn't complaining about it ?

Because we didn't have a `rustfmt.toml` before.

File: packable/packable/CHANGELOG.md [hunk 3]
```diff
@@ -19,8 +19,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+
+## 0.X.X - 2022-XX-XX
+
 ## 0.2.1 - 2022-02-11
 
+### Changed
+
+- Make byte buffers packing and unpacking more performant.
+
```
Comment (thibault-martinez) [inline packable/packable/CHANGELOG.md:30]: This version was already released

File: rustfmt.toml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+version = "Two"
+comment_width = 120
+format_code_in_doc_comments = true
+max_width = 120
+imports_granularity = "Crate"
+normalize_comments = true
+normalize_doc_attributes = true
+wrap_comments = true
```
Comment (thibault-martinez) [inline rustfmt.toml:8]: Do we want to add the license thingy ?

Comment (pvdrz) [reply rustfmt.toml:8]: up to you

File: packable/packable/CHANGELOG.md (hunk 3 above)
Comment (pvdrz) [reply packable/packable/CHANGELOG.md:28]: wrong version section. I fixed it in the last commit

File: rustfmt.toml (hunk 4 above)
Comment (thibault-martinez) [reply rustfmt.toml:8]: Yes please

File: packable/packable/CHANGELOG.md [hunk 5]
```diff
@@ -19,6 +19,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+
+## 0.X.X - 2022-XX-XX
+
+### Changed
+
+- Make byte buffers packing and unpacking more performant.
```
Comment (thibault-martinez) [inline packable/packable/CHANGELOG.md:27]:
```suggestion
- Make byte buffers packing and unpacking more performant;
```

File: rustfmt.toml (hunk 4 above)
Comment (pvdrz) [reply rustfmt.toml:8]: done :p

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**